src/platform/plugins/shared/discover/public/application/main/state_management/discover_data_state_container.ts

```diff
@@ -349,17 +349,14 @@ export function getDataStateContainer({
 
           abortController = new AbortController();
 
-          const isEsqlQuery = isOfAggregateQueryType(getCurrentTab().appState.query);
           const latestFetchDetails: DiscoverLatestFetchDetails = {
             abortController,
           };
 
           // Trigger chart fetching after the pre fetch state has been updated
           // to ensure state values that would affect data fetching are set
-          if (!isEsqlQuery) {
-            // trigger in parallel with the main request for Classic mode
-            fetchChart$.next(latestFetchDetails);
-          }
+          // trigger in parallel with the main request for Classic mode
+          fetchChart$.next(latestFetchDetails);
```

Comment:
So this is about where I'd expect `esqlQueryColumns` to be updated using `getESQLQueryColumns`, probably right above where `fetchChart$` is triggered so it's updated by the time chart fetching occurs. Currently we update it here after the main fetch, which is the piece that I think should change (just `esqlQueryColumns`, the rest should remain the same): https://github.com/elastic/kibana/blob/27c0e7ddebc444512b2a6118c180c9398035c69f/src/platform/plugins/shared/discover/public/application/main/data_fetching/fetch_all.ts#L180-L187

Removing the conditionals around `fetchChart$` seems right though since we'd then always have the columns by the time this line runs.